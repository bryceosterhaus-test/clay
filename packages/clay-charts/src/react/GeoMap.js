import React from 'react';
import {Geomap} from '../util/geo-map';

export default class GeomapReact extends React.Component {
	constructor(props) {
		super(props);

		this._containerRef = React.createRef();
	}
	componentDidMount() {
		this._geoMapInstance = new Geomap({
			...this.props,
			element: this._containerRef.current,
		});

		this._geoMapInstance.attached();
	}

	/**
	 * @inheritDoc
	 */
	componentWillUnmount() {
		this._geoMapInstance.disposed();
	}

	render() {
		const {data, element, color, pollingInterval, ...otherProps} = this.props;

		const {height = '100%', width = '100%'} = this._geoMapInstance
			? this._geoMapInstance.getSize()
			: {};

		return (
			<div
				style={{height, width}}
				{...otherProps}
				ref={this._containerRef}
			/>
		);
	}
}
