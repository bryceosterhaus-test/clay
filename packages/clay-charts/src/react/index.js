import React from 'react';
import BillboardChart from 'react-billboardjs';
import GeoMap from './react/GeoMap';

import {
	DEFAULT_COLORS,
	DEFAULT_GRID_OBJECT,
	DEFAULT_LINE_CLASSES,
	DEFAULT_POINT_PATTERNS,
} from '../ChartBase';

import '../scss/main.scss';

export default class extends React.Component {
	render() {
		const {color, data, grid, line, point, ...otherProps} = this.props;

		let ChartComponent;
		let props = {data, ...otherProps};

		switch (data.type) {
			case 'geo-map':
				ChartComponent = GeoMap;
				break;
			default:
				ChartComponent = BillboardChart;
		}

		return (
			<ChartComponent
				{...otherProps}
				data={data}
				grid={Object.assign(DEFAULT_GRID_OBJECT, grid)}
				color={Object.assign({pattern: DEFAULT_COLORS}, color)}
				line={Object.assign({classes: DEFAULT_LINE_CLASSES}, line)}
				point={Object.assign(
					{
						pattern: DEFAULT_POINT_PATTERNS,
					},
					point
				)}
			/>
		);
	}
}
