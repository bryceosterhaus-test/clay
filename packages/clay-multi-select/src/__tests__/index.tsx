/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import ClayMultiSelect, {ClayMultiSelectWithAutocomplete} from '..';
import {cleanup, fireEvent, render} from 'react-testing-library';

describe('ClayMultiSelect', () => {
	it('renders', () => {
		const testRenderer = TestRenderer.create(
			<ClayMultiSelect
				items={[]}
				onItemsChange={() => {}}
				spritemap="/foo/bar"
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});

	it('renders with items', () => {
		const testRenderer = TestRenderer.create(
			<ClayMultiSelect
				items={['foo', 'bar', 'baz']}
				onItemsChange={() => {}}
				spritemap="/foo/bar"
			/>
		);

		expect(testRenderer.toJSON()).toMatchSnapshot();
	});
});

const ClayMultiSelectWithState = (props: any) => {
	const [items, setItems] = React.useState([]);

	return (
		<ClayMultiSelect
			{...props}
			items={items}
			onItemsChange={setItems}
			spritemap="/foo/bar"
		/>
	);
};

describe('Interactions', () => {
	afterEach(cleanup);

	it('adding text and hitting `,` should add a new items', () => {
		const {container} = render(<ClayMultiSelectWithState />);

		const input = container.querySelector('input');

		fireEvent.change(input as HTMLInputElement, {
			target: {value: 'foo'},
		});

		fireEvent.keyDown(input as HTMLInputElement, {
			keyCode: 188,
		});

		expect(container.querySelectorAll('.label').length).toEqual(1);

		fireEvent.change(input as HTMLInputElement, {
			target: {value: 'bar'},
		});

		fireEvent.keyDown(input as HTMLInputElement, {keyCode: 188});

		expect(container.querySelectorAll('.label').length).toEqual(2);
	});

	it('if value does not pass validation, it should not turn into a label', () => {
		const {container} = render(
			<ClayMultiSelectWithState
				validationFn={(val: string) => val === 'foo'}
			/>
		);

		const input = container.querySelector('input');

		fireEvent.change(input as HTMLInputElement, {
			target: {value: 'foo'},
		});

		fireEvent.keyDown(input as HTMLInputElement, {
			keyCode: 188,
		});

		expect(container.querySelectorAll('.label').length).toEqual(1);

		fireEvent.change(input as HTMLInputElement, {
			target: {value: 'bar'},
		});

		fireEvent.keyDown(input as HTMLInputElement, {keyCode: 188});

		expect(container.querySelectorAll('.label').length).toEqual(1);
	});

	it('clicking x removes item', () => {
		const onItemsChangeFn = jest.fn();

		const {container} = render(
			<ClayMultiSelect
				items={['foo']}
				onItemsChange={onItemsChangeFn}
				spritemap="/foo/bar"
			/>
		);

		expect(container.querySelectorAll('.label').length).toEqual(1);

		expect(onItemsChangeFn).not.toHaveBeenCalled();

		fireEvent.click(
			container.querySelector('.label .close') as HTMLButtonElement,
			{}
		);

		expect(onItemsChangeFn).toHaveBeenCalled();
	});

	it('clicking on autocomplete item adds calls onItemsChange', () => {
		const onItemsChangeFn = jest.fn();

		render(
			<ClayMultiSelectWithAutocomplete
				inputValue="two"
				items={['foo']}
				onItemsChange={onItemsChangeFn}
				sourceItems={['one', 'two', 'three', 'four']}
				spritemap="/foo/bar"
			/>
		);

		expect(document.querySelectorAll('.dropdown-item').length).toEqual(1);

		expect(onItemsChangeFn).not.toHaveBeenCalled();

		fireEvent.click(
			document.querySelectorAll('.dropdown-item')[0] as HTMLLIElement,
			{}
		);

		expect(onItemsChangeFn).toHaveBeenCalled();
	});
});
