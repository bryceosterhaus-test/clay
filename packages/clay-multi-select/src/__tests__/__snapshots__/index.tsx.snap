// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayMultiSelect renders 1`] = `
<div
  className="form-group"
>
  <div
    className="input-group input-group-stacked-sm-down"
  >
    <div
      className="input-group-item"
    >
      <div
        className="form-control form-control-tag-group"
      >
        <input
          className="form-control-inset"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ClayMultiSelect renders with items 1`] = `
<div
  className="form-group"
>
  <div
    className="input-group input-group-stacked-sm-down"
  >
    <div
      className="input-group-item"
    >
      <div
        className="form-control form-control-tag-group"
      >
        <span
          className="label label-dismissible label-secondary"
        >
          <span
            className="label-item label-item-expand"
          >
            foo
          </span>
          <span
            className="label-item label-item-after"
          >
            <button
              className="close"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="lexicon-icon lexicon-icon-times"
                role="presentation"
              >
                <use
                  xlinkHref="/foo/bar#times"
                />
              </svg>
            </button>
          </span>
        </span>
        <span
          className="label label-dismissible label-secondary"
        >
          <span
            className="label-item label-item-expand"
          >
            bar
          </span>
          <span
            className="label-item label-item-after"
          >
            <button
              className="close"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="lexicon-icon lexicon-icon-times"
                role="presentation"
              >
                <use
                  xlinkHref="/foo/bar#times"
                />
              </svg>
            </button>
          </span>
        </span>
        <span
          className="label label-dismissible label-secondary"
        >
          <span
            className="label-item label-item-expand"
          >
            baz
          </span>
          <span
            className="label-item label-item-after"
          >
            <button
              className="close"
              onClick={[Function]}
              type="button"
            >
              <svg
                className="lexicon-icon lexicon-icon-times"
                role="presentation"
              >
                <use
                  xlinkHref="/foo/bar#times"
                />
              </svg>
            </button>
          </span>
        </span>
        <input
          className="form-control-inset"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onPaste={[Function]}
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
